cmake/x64-windows-d3.cmake

```diff
@@ -0,0 +1,12 @@
+# Custom triplet for vcpkg that builds some packages statically
+
+set(VCPKG_TARGET_ARCHITECTURE x64)
+
+# Link statically selected ports
+if(${PORT} MATCHES "zlib")
+  set(VCPKG_CRT_LINKAGE static)
```

Comment:
We build all projects against the dynamic C runtime (compiler option /MD), we can't mix that with a static C runtime (/MT). So `set(VCPKG_CRT_LINKAGE dynamic)` should be used for static dependencies, too.

Reply:
This is not building project statically itself, it's about statically building external dependency (zlib), so we can compile-in statically built zlib into executable. Application itself compiles with /MD option.

Reply:
I know, but that's not the point. All dependencies need to use the same CRT, ergo the same compiler option for either /MD **or** /MT. It doesn't matter if it's our own code or some external dependency. You can still build zlib as a static lib, but it needs to use the dynamic CRT as everything else in our project.

Reply:
I don't follow, why I'm not allow to do what I want to with own code. This PR compiles fine and no any issues regarding static and dynamic linking are present. Resulting binary has statically linked zlib and runs without external DLL. Can you explain please, what is issue here?

Reply:
This blog post gives some info. https://www.caichinger.com/blog/2016/01/01/mixing_crt_versions/
It's safe (within reasonable boundaries) for a DLL to use a different CRT than the program that links to it, but not for a statically-linked library.

Reply:
@pzychotic @winterheart How should we settle this? I'm no Windows compilation expert, but compiling zlib statically to avoid having to copy over a DLL would improve developer and user experience

Reply:
We can link zlib statically there is no problem with that. We should only change the CRT linkage to dynamic for that also.

I'm still on vacation so I can't test this, but the change needed would be to remove the line `set(VCPKG_CRT_LINKAGE static)` from the if block and move the `set(VCPKG_CRT_LINKAGE dynamic)` outside of the else block.

Reply:
@winterheart could you try this out ? :)